### DISCUSSION TITLE
Change Request "Preview" Does Not Reflect Latest Changes

### DISCUSSION THREAD

**Discussion by nikkisaildrone** [GitbookIO/community#575] [0. General bugs and questions]
When working with the Gitbook editor, I've noticed that the "Preview" site does not reflect the latest revisions. It seems to reflect only revisions from the first time "Preview" is triggered. 

Here's the preview of my site:
![Screenshot 2024-04-10 at 5 48 47 PM](https://github.com/GitbookIO/community/assets/85899726/f74b3491-36b8-41e5-9a01-782e6cb486a0)

But here is the current state with additional revisions that are not reflected in the "Preview":

![Screenshot 2024-04-10 at 5 49 03 PM](https://github.com/GitbookIO/community/assets/85899726/f316dead-8713-4778-badb-a205c8c3336c)

I've been able to consistently reproduce this with several change requests so it seems like possibly a bug?

**Comment by aikoii** [comment]
Hi @nikkisaildrone thanks for flagging! We'll look into it 🙏

**Comment by npatki** [comment]
I have been observing this too, ever since I started generating Previews. Didn't realize it was a bug. I thought maybe it wasn't a supported feature yet. 

My workflow: I am in the habit of generating preview links to share with other folks in my company. If I update my change request based on their feedback, they are unable to see my changes. This is causing an impediment in our workflow.

**Comment by nikkisaildrone** [comment]
We have a similar workflow as we're starting to develop content for our user docs. Have been working with the PM to review his updates in a change request, and agree this limitation has been an impediment.

It's even more frustrating when we're trying to preview things like the OpenAPI docs and discovering what's rendered in the WYSIWYG editor isn't necessarily 1:1 with the published content. Things like the "Test it" feature and code previews get rendered only with the publish/preview event. So having that Preview work reliably is important when editing those blocks.

**Comment by islamellawendy** [comment]
I am still having this issue too while trying to preview things in my CR, The first update only shows and a whole page does not show.

Also, I noticed that Snippets is not showing in preview as it supposed to in preview, could it be same related thing or a different bug? (Considering that it is beta)

**Comment by waltergalvao** [comment]
I also have the same problem, and this is a suuuper annoying and blocking bug. It kills the whole "review requests" feature, as reviewers can't see latest changes live. Please fix this, it has been months 😢.

@aikoii @emmerich @addisonschultz @SamyPesse

**Comment by krautech** [comment]
Yeah ive also encountered this. Its really annoying

**Comment by dangnelson** [comment]
Yeah this is incredibly annoying. Just to see the changes I just made I have to create a new change request. So it means I have a ton of blank change requests with no content that is actually changed.

**Comment by aikoii** [comment]
Good news! We've fixed Preview updates 🎉. There _might_ be a few seconds delay now, but it should be much faster.

You'll also notice that the toolbar looks nicer with more buttons, which will give you more info about your changes.

![preview](https://github.com/user-attachments/assets/58299a2f-d14e-4a80-81d3-928007159bc3)

Thanks for hanging in there while we made these improvements 💙
